File: Ent_UsAdmin/TraspasarDocumentosSolicitud.vb
Imports System.ComponentModel
Imports System.Data.SqlClient
Imports Microsoft.Office.Interop

Public Class TraspasarDocumentosSolicitud
    Public ipOrigen As String
    Public bdOrigen As String
    Public cnOrigen As String
    Dim conexionOrigen As SqlClient.SqlConnection
    Public ipDestino As String
    Public bdDestino As String
    Dim cnDestino As String
    Dim conexionDestino As SqlClient.SqlConnection
    Dim dataDocumentosSolicitud As DataTable
    Dim adapterDocumentosSolicitud As SqlDataAdapter
    Dim ncargando As Cargando
    Public fila As String
    Public columna As String
    Dim tipo As String


    Private Sub TraspasarDocumentosSolicitud_Load(sender As Object, e As EventArgs) Handles MyBase.Load

    End Sub
    Private Sub iniciarconexionOrigen()
        cnOrigen = "Data Source=  " & ipOrigen & "\confia;" &
                     "Initial Catalog=" & bdOrigen & ";" &
                     "User Id=sa;Password=BSi5t3Ma$CFAD;MultipleActiveResultSets=true"
        conexionOrigen = New SqlConnection(cnOrigen)
        conexionOrigen.Open()
    End Sub
    Private Sub iniciarconexionDestino()
        cnDestino = "Data Source=  " & ipDestino & "\confia;" &
                     "Initial Catalog=" & bdDestino & ";" &
                     "User Id=sa;Password=BSi5t3Ma$CFAD;MultipleActiveResultSets=true"
        conexionDestino = New SqlConnection(cnDestino)
        conexionDestino.Open()
    End Sub

    Private Sub btnOrigen_Click(sender As Object, e As EventArgs) Handles btnOrigen.Click
        EmpresasOrigenDocSolicitud.ShowDialog()

    End Sub

    Private Sub btnDestino_Click(sender As Object, e As EventArgs) Handles btnDestino.Click
        EmpresasDestinoDocSolicitud.ShowDialog()

    End Sub

    Private Sub BackgroundWorker1_DoWork(sender As Object, e As System.ComponentModel.DoWorkEventArgs) Handles BackgroundWorker1.DoWork
        iniciarconexionOrigen()

        adapterDocumentosSolicitud = New SqlDataAdapter(txtSQL.Text, conexionOrigen)
        dataDocumentosSolicitud = New DataTable
        adapterDocumentosSolicitud.Fill(dataDocumentosSolicitud)

    End Sub

    Private Sub btnConsulta_Click(sender As Object, e As EventArgs) Handles btnConsulta.Click
        ncargando = New Cargando
        ncargando.Show()
        ncargando.MonoFlat_Label1.Text = "Consultando"
        BackgroundWorker1.RunWorkerAsync()

    End Sub

    Private Sub BackgroundWorker1_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundWorker1.RunWorkerCompleted
        ncargando.Close()

    End Sub

    Private Sub TextBox1_TextChanged(sender As Object, e As EventArgs) Handles TextBox1.TextChanged

    End Sub

    Private Sub TextBox1_KeyDown(sender As Object, e As KeyEventArgs) Handles TextBox1.KeyDown
        If e.KeyCode = Keys.F2 Then
            Dim abrircarpeta As OpenFileDialog
            abrircarpeta = New OpenFileDialog
            abrircarpeta.Filter = "Archivo de Excel|*.xls;*.xlsx"
            If abrircarpeta.ShowDialog = Windows.Forms.DialogResult.OK Then
                TextBox1.Text = abrircarpeta.FileName
            Else
                MessageBox.Show("No se seleccionó ningún archivo")
            End If
            ImportarExcelTraspaso.ShowDialog()

        End If
    End Sub

    Private Sub btnCargar_Click(sender As Object, e As EventArgs) Handles btnCargar.Click
        Try
            ncargando = New Cargando

            ncargando.Show()
            ncargando.MonoFlat_Label1.Text = "Cargando datos"
            Dim intLoopCounter As Integer
            Dim contador1 As Integer
            Dim contador As Integer = 1
            Dim cantidadregistros As Integer = 1

            Dim fname As String
            Dim objXLApp As Excel.Application
            objXLApp = New Excel.Application
            fname = TextBox1.Text
            With objXLApp
                .Workbooks.Open(fname)
                .Workbooks(1).Worksheets(1).Select()

                For intLoopCounter = fila To CInt(.ActiveSheet.Cells.SpecialCells(Excel.XlCellType.xlCellTypeLastCell).Row)
                    If Convert.ToString(.Range("" & columna & "" & intLoopCounter).Value) = "" Then
                        Exit For


                    End If

                    cantidadregistros += 1

                Next intLoopCounter

                For intLoopCounter = fila To cantidadregistros + fila - 2
                    'panelespera.Visible = True

                    '  BunifuCircleProgressbar1.MaxValue = cantidadregistros + fila - 2
                    ncargando.MonoFlat_Label1.Text = "" & contador & " registros de " & cantidadregistros - 1 & ""
                    '  BunifuCircleProgressbar1.Value = intLoopCounter
                    ' Sql = "insert into hotel(Nombre,telefono,email) values('" & .Range("" & Combonombre.Text & "" & intLoopCounter) & "','" & .Range("" & Combotelefono.Text & "" & intLoopCounter) & "','" & .Range("" & Comboemail.Text & "" & intLoopCounter) & "')"

                    dtDatos.Rows.Add(.Range("" & columna & "" & intLoopCounter).Value, .Range("" & columna & "" & intLoopCounter).Offset(0, 1).Value, .Range("" & columna & "" & intLoopCounter).Offset(0, 2).Value, .Range("" & columna & "" & intLoopCounter).Offset(0, 3).Value)

                    contador += 1
                Next intLoopCounter
                .Workbooks(1).Close(False)
                .Quit()
            End With

        Catch ex As Exception
            MessageBox.Show(ex.Message)
        End Try
        ncargando.Close()
    End Sub

    Private Sub BackgroundTraspaso_DoWork(sender As Object, e As DoWorkEventArgs) Handles BackgroundTraspaso.DoWork
        iniciarconexionDestino()
        If tipo = "Solicitud" Then
            For Each row As DataRow In dataDocumentosSolicitud.Rows
                For Each rowid As DataGridViewRow In dtDatos.Rows
                    If row("IdDatosSolicitud").ToString = rowid.Cells(0).Value Then
                        Dim comandoDocumento As SqlCommand
                        Dim consultaDocumento As String
                        consultaDocumento = "insert into documentossolicitud values('" & rowid.Cells(1).Value & "','" & row("Tipo") & "',@imagen)"
                        Dim imgDocumento As New SqlParameter("@imagen", SqlDbType.Image)
                        imgDocumento.Value = row("Imagen")
                        comandoDocumento = New SqlCommand
                        comandoDocumento.Connection = conexionDestino
                        comandoDocumento.CommandText = consultaDocumento
                        comandoDocumento.Parameters.Add(imgDocumento)
                        comandoDocumento.ExecuteNonQuery()


                    End If
                Next
            Next
        Else
            For Each row As DataRow In dataDocumentosSolicitud.Rows
                For Each rowid As DataGridViewRow In dtDatos.Rows
                    If row("IdCredito").ToString = rowid.Cells(0).Value Then
                        Dim comandoDocumento As SqlCommand
                        Dim consultaDocumento As String
                        consultaDocumento = "insert into documentosCredito values('" & rowid.Cells(1).Value & "','" & row("Tipo") & "',@imagen)"
                        Dim imgDocumento As New SqlParameter("@imagen", SqlDbType.Image)
                        imgDocumento.Value = row("Imagen")
                        comandoDocumento = New SqlCommand
                        comandoDocumento.Connection = conexionDestino
                        comandoDocumento.CommandText = consultaDocumento
                        comandoDocumento.Parameters.Add(imgDocumento)
                        comandoDocumento.ExecuteNonQuery()


                    End If
                Next
            Next
        End If


    End Sub

    Private Sub BackgroundTraspaso_RunWorkerCompleted(sender As Object, e As RunWorkerCompletedEventArgs) Handles BackgroundTraspaso.RunWorkerCompleted
        ncargando.Close()

    End Sub

    Private Sub btnTraspasar_Click(sender As Object, e As EventArgs) Handles btnTraspasar.Click
        ncargando = New Cargando
        ncargando.Show()
        ncargando.MonoFlat_Label1.Text = "Realizando traspaso de documentos"
        BackgroundTraspaso.RunWorkerAsync()

    End Sub

    Private Sub RadioSolicitud_CheckedChanged(sender As Object, e As EventArgs) Handles RadioSolicitud.CheckedChanged
        If RadioSolicitud.Checked Then
            tipo = "Solicitud"
        Else
            tipo = "Crédito"
        End If
    End Sub

    Private Sub RadioCredito_CheckedChanged(sender As Object, e As EventArgs) Handles RadioCredito.CheckedChanged
        If RadioCredito.Checked Then
            tipo = "Crédito"
        Else
            tipo = "Solicitud"

        End If
    End Sub
End Class

File: Ent_UsAdmin/Solicitud/TraspasarDocumentosSolicitud.cs
using System;
using System.ComponentModel;
using System.Data;
using System.Data.SqlClient;
using System.Windows.Forms;
using Microsoft.VisualBasic.CompilerServices;

namespace ConfiaAdmin
{

    public partial class TraspasarDocumentosSolicitud
    {
        public string ipOrigen;
        public string bdOrigen;
        public string cnOrigen;
        private SqlConnection conexionOrigen;
        public string ipDestino;
        public string bdDestino;
        private string cnDestino;
        private SqlConnection conexionDestino;
        private DataTable dataDocumentosSolicitud;
        private SqlDataAdapter adapterDocumentosSolicitud;
        private Cargando ncargando;
        public string fila;
        public string columna;
        private string tipo;

        public TraspasarDocumentosSolicitud()
        {
            InitializeComponent();
        }


        private void TraspasarDocumentosSolicitud_Load(object sender, EventArgs e)
        {

        }
        private void iniciarconexionOrigen()
        {
            cnOrigen = "Data Source=  " + ipOrigen + @"\confia;" + "Initial Catalog=" + bdOrigen + ";" + "User Id=sa;Password=BSi5t3Ma$CFAD;MultipleActiveResultSets=true";
            conexionOrigen = new SqlConnection(cnOrigen);
            conexionOrigen.Open();
        }
        private void iniciarconexionDestino()
        {
            cnDestino = "Data Source=  " + ipDestino + @"\confia;" + "Initial Catalog=" + bdDestino + ";" + "User Id=sa;Password=BSi5t3Ma$CFAD;MultipleActiveResultSets=true";
            conexionDestino = new SqlConnection(cnDestino);
            conexionDestino.Open();
        }

        private void btnOrigen_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.EmpresasOrigenDocSolicitud.ShowDialog();

        }

        private void btnDestino_Click(object sender, EventArgs e)
        {
            My.MyProject.Forms.EmpresasDestinoDocSolicitud.ShowDialog();

        }

        private void BackgroundWorker1_DoWork(object sender, DoWorkEventArgs e)
        {
            iniciarconexionOrigen();

            adapterDocumentosSolicitud = new SqlDataAdapter(txtSQL.Text, conexionOrigen);
            dataDocumentosSolicitud = new DataTable();
            adapterDocumentosSolicitud.Fill(dataDocumentosSolicitud);

        }

        private void btnConsulta_Click(object sender, EventArgs e)
        {
            ncargando = new Cargando();
            ncargando.Show();
            ncargando.MonoFlat_Label1.Text = "Consultando";
            BackgroundWorker1.RunWorkerAsync();

        }

        private void BackgroundWorker1_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            ncargando.Close();

        }

        private void TextBox1_TextChanged(object sender, EventArgs e)
        {

        }

        private void TextBox1_KeyDown(object sender, KeyEventArgs e)
        {
            if (e.KeyCode == Keys.F2)
            {
                OpenFileDialog abrircarpeta;
                abrircarpeta = new OpenFileDialog();
                abrircarpeta.Filter = "Archivo de Excel|*.xls;*.xlsx";
                if (abrircarpeta.ShowDialog() == DialogResult.OK)
                {
                    TextBox1.Text = abrircarpeta.FileName;
                }
                else
                {
                    MessageBox.Show("No se seleccionó ningún archivo");
                }
                My.MyProject.Forms.ImportarExcelTraspaso.ShowDialog();

            }
        }

        private void btnCargar_Click(object sender, EventArgs e)
        {
            try
            {
                ncargando = new Cargando();

                ncargando.Show();
                ncargando.MonoFlat_Label1.Text = "Cargando datos";
                int intLoopCounter;
                int contador1;
                int contador = 1;
                int cantidadregistros = 1;

                string fname;
                Microsoft.Office.Interop.Excel.Application objXLApp;
                objXLApp = new Microsoft.Office.Interop.Excel.Application();
                fname = TextBox1.Text;
                objXLApp.Workbooks.Open(fname);
                objXLApp.Workbooks[1].Worksheets[(object)1].Select();

                var loopTo = Conversions.ToInteger(objXLApp.ActiveSheet.Cells.SpecialCells(Microsoft.Office.Interop.Excel.XlCellType.xlCellTypeLastCell).Row);
                for (intLoopCounter = Conversions.ToInteger(fila); intLoopCounter <= loopTo; intLoopCounter++)
                {
                    if (string.IsNullOrEmpty(Convert.ToString(objXLApp.get_Range("" + columna + "" + intLoopCounter).get_Value())))
                    {
                        break;


                    }

                    cantidadregistros += 1;

                }

                var loopTo1 = (int)Math.Round(cantidadregistros + Conversions.ToDouble(fila) - 2d);
                for (intLoopCounter = Conversions.ToInteger(fila); intLoopCounter <= loopTo1; intLoopCounter++)
                {
                    // panelespera.Visible = True

                    // BunifuCircleProgressbar1.MaxValue = cantidadregistros + fila - 2
                    ncargando.MonoFlat_Label1.Text = "" + contador + " registros de " + (cantidadregistros - 1) + "";
                    // BunifuCircleProgressbar1.Value = intLoopCounter
                    // Sql = "insert into hotel(Nombre,telefono,email) values('" & .Range("" & Combonombre.Text & "" & intLoopCounter) & "','" & .Range("" & Combotelefono.Text & "" & intLoopCounter) & "','" & .Range("" & Comboemail.Text & "" & intLoopCounter) & "')"

                    dtDatos.Rows.Add(objXLApp.get_Range("" + columna + "" + intLoopCounter).get_Value(), objXLApp.get_Range("" + columna + "" + intLoopCounter).get_Offset(0, 1).get_Value(), objXLApp.get_Range("" + columna + "" + intLoopCounter).get_Offset(0, 2).get_Value(), objXLApp.get_Range("" + columna + "" + intLoopCounter).get_Offset(0, 3).get_Value());

                    contador += 1;
                }
                objXLApp.Workbooks[1].Close(false);
                objXLApp.Quit();
            }

            catch (Exception ex)
            {
                MessageBox.Show(ex.Message);
            }
            ncargando.Close();
        }

        private void BackgroundTraspaso_DoWork(object sender, DoWorkEventArgs e)
        {
            iniciarconexionDestino();
            if (tipo == "Solicitud")
            {
                foreach (DataRow row in dataDocumentosSolicitud.Rows)
                {
                    foreach (DataGridViewRow rowid in dtDatos.Rows)
                    {
                        if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(row["IdDatosSolicitud"].ToString(), rowid.Cells[0].Value, false)))
                        {
                            SqlCommand comandoDocumento;
                            string consultaDocumento;
                            consultaDocumento = Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject("insert into documentossolicitud values('", rowid.Cells[1].Value), "','"), row["Tipo"]), "',@imagen)"));
                            var imgDocumento = new SqlParameter("@imagen", SqlDbType.Image);
                            imgDocumento.Value = row["Imagen"];
                            comandoDocumento = new SqlCommand();
                            comandoDocumento.Connection = conexionDestino;
                            comandoDocumento.CommandText = consultaDocumento;
                            comandoDocumento.Parameters.Add(imgDocumento);
                            comandoDocumento.ExecuteNonQuery();


                        }
                    }
                }
            }
            else
            {
                foreach (DataRow row in dataDocumentosSolicitud.Rows)
                {
                    foreach (DataGridViewRow rowid in dtDatos.Rows)
                    {
                        if (Conversions.ToBoolean(Operators.ConditionalCompareObjectEqual(row["IdCredito"].ToString(), rowid.Cells[0].Value, false)))
                        {
                            SqlCommand comandoDocumento;
                            string consultaDocumento;
                            consultaDocumento = Conversions.ToString(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject(Operators.ConcatenateObject("insert into documentosCredito values('", rowid.Cells[1].Value), "','"), row["Tipo"]), "',@imagen)"));
                            var imgDocumento = new SqlParameter("@imagen", SqlDbType.Image);
                            imgDocumento.Value = row["Imagen"];
                            comandoDocumento = new SqlCommand();
                            comandoDocumento.Connection = conexionDestino;
                            comandoDocumento.CommandText = consultaDocumento;
                            comandoDocumento.Parameters.Add(imgDocumento);
                            comandoDocumento.ExecuteNonQuery();


                        }
                    }
                }
            }


        }

        private void BackgroundTraspaso_RunWorkerCompleted(object sender, RunWorkerCompletedEventArgs e)
        {
            ncargando.Close();

        }

        private void btnTraspasar_Click(object sender, EventArgs e)
        {
            ncargando = new Cargando();
            ncargando.Show();
            ncargando.MonoFlat_Label1.Text = "Realizando traspaso de documentos";
            BackgroundTraspaso.RunWorkerAsync();

        }

        private void RadioSolicitud_CheckedChanged(object sender, EventArgs e)
        {
            if (RadioSolicitud.Checked)
            {
                tipo = "Solicitud";
            }
            else
            {
                tipo = "Crédito";
            }
        }

        private void RadioCredito_CheckedChanged(object sender, EventArgs e)
        {
            if (RadioCredito.Checked)
            {
                tipo = "Crédito";
            }
            else
            {
                tipo = "Solicitud";

            }
        }
    }
}